internal/controller/factory/etcd_client.go

```
@@ -29,6 +33,7 @@
 		cfg.Endpoints = []string{ep}
 		singleClients[i], err = clientv3.New(cfg)
 		if err != nil {
+			clusterClient.Close()
```

golangci-lint (1.64.8)

Error return value of `clusterClient.Close` is not checked

(errcheck)